### PR TITLE
[e2e-tests]Use nodeAffinity instead of labeling nodes

### PR DESCRIPTION
**Pull request by Barakmor1**
This will make tests more readable and prevent
modifying nodes.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/swap_test.go [hunk 1]
```diff
@@ -224,27 +222,26 @@ var _ = Describe("[Serial][sig-compute]SwapTest", Serial, decorators.SigCompute,
...
 			vmiMemSize := resource.MustParse(fmt.Sprintf("%dMi", int((float64(memToUseInTargetNodeVmKib)+float64(gigbytesInkib*2))/bytesInKib)))
 			vmiMemReq := resource.MustParse(fmt.Sprintf("%dMi", vmMemoryRequestkib/bytesInKib))
 			vmiToFillTargetNodeMem := tests.NewRandomFedoraVMIWithGuestAgent()
-			vmiToFillTargetNodeMem.Spec.NodeSelector = map[string]string{"swaptest": "swaptest"}
+			//we want vmiToFillTargetNodeMem to land on the target node to achieve memory-overcommitment in target
+			nodeAffinityRuleForVmiToFill, err := getAffinityForTargetNode(targetNode)
+			Expect(err).ToNot(HaveOccurred())
+			vmiToFillTargetNodeMem.Spec.Affinity = &v1.Affinity{
+				NodeAffinity: nodeAffinityRuleForVmiToFill,
+			}
```
Comment (jean-edouard) [inline tests/swap_test.go:244]: Nit: maybe `getAffinityForTargetNode()` could return a `*v1.Affinity` so its return value could be used directly?

**Review by jean-edouard — COMMENTED** [review]
/approve
Just one nit in-line.

File: tests/swap_test.go (hunk 1 above)
Comment (Barakmor1) [reply tests/swap_test.go:244]: Done

**Comment by iholder101** [comment]
Looks much better, thanks!
/lgtm

**Pull request merged**